modules/launcher/src/main/java/org/mule/runtime/module/launcher/MuleContainer.java

```diff
@@ -217,6 +235,52 @@ protected void init(String[] args) throws IllegalArgumentException, Initialisati
     artifactResourcesRegistry.getMemoryManagementService().initialise();
     artifactResourcesRegistry.inject(artifactResourcesRegistry.getContainerProfilingService());
     initialiseIfNeeded(artifactResourcesRegistry.getContainerProfilingService());
+
+    if (isFipsEnabled()) {
+      configureSecurityManager();
+      setSecurityAlgorithm();
+    }
+  }
+
+  private void configureSecurityManager() throws InitialisationException {
+    try {
+      Class<?> classDef =
+          Class.forName(System.getProperty(PROPERTY_FIPS_PROVIDER, "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider"));
+      Constructor<?> constructor = classDef.getConstructor();
+      Provider fipsProvider = (Provider) constructor.newInstance();
+      java.security.Security.insertProviderAt(fipsProvider, 1);
+
+      classDef =
+          Class.forName(System.getProperty(PROPERTY_JSSE_PROVIDER, "org.bouncycastle.jsse.provider.BouncyCastleJsseProvider"));
+      constructor = classDef.getConstructor();
+      Provider jsseProvider = (Provider) constructor.newInstance();
+      jsseProvider.setProperty(FIPS_KEY, FIPS_VALUE);
+      java.security.Security.insertProviderAt(jsseProvider, 2);
+
+      Provider sunJsseProvider = Security.getProvider(SUN_JSSE_PROVIDER);
+      // for java 8
+      Provider sunJsseProviderLegacy = Security.getProvider("com.sun.net.ssl.internal.ssl.Provider");
+
+      if (sunJsseProvider != null) {
+        Security.removeProvider(SUN_JSSE_PROVIDER);
+      }
+
+      if (sunJsseProviderLegacy != null) {
+        sunJsseProviderLegacy.setProperty(FIPS_VALUE, "");
+      }
+    } catch (ClassNotFoundException | InvocationTargetException | NoSuchMethodException | InstantiationException
+        | IllegalAccessException e) {
+      LOGGER.error("Critical error while enabling FIPS:" + e.getMessage());
```

Comment:
I think it is better to include the cause for the logged error message instead of concatenating just the `e.getMessage()`.
```suggestion
      LOGGER.error("Critical error while enabling FIPS", e);
```